rfcs/0000-api-cr-crowding.md

```diff
@@ -0,0 +1,337 @@
+  only relevant occupancies via route, trip, or trip short name is necessary.
+- `?include=trip`. Maybe filtering isn't necessary, but being able to include the trip, in order
+  to have easy access to the trip short name, is necessary.
+
+But for the time being, we won't allow other resources to include occupancies. For example, we
+don't want clients to do something like `/trips?filter[route]=CR-Worcester&include=occupancies`.
+In the event that something substantial changes about occupancies (or if it's removed!), the
+changes are limited to just this one resource, which is a risk of using the experimental resource.
+On the other hand, if this proposal matures and we build forth, clients who have built
+interactions with this endpoint will continue to have this as a supported method of access.
+
+# Reference-level explanation
+
+<!--[reference-level-explanation]: #reference-level-explanation
+
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how
+the detailed proposal makes those examples work. -->
+
+## Keolis data
+
+The Keolis data comes from a firebase feed. It looks like:
+
+```
+{
+  "data": [
+    {
+      "cTrainNo": "003     ",
+      "MedianDensity": 0.58635,
+      "MedianDensityFlag": 1
+    },
+    //...
+  ],
+  "last_updated": "2021-09-24 15:30:00"
+}
+```
+
+Each record in `"data"` is for one trip, with `"cTrainNo"` being the train's name (with extraneous
+padding for arcane reasons on Keolis's end).
+
+The `"MedianDensity"` is the percentage of seats occupied, and `"MedianDensityFlag` is an enum
+with values `0`, `1`, or `2`, representing Keolis's idea of "Not Crowded", "Some Crowding", and
+"Crowded" respectively.
+
+## API data
+
+In the API, we'll add a new kind of model `Occupancy` with its own `State.Server`. The model will
+have keys (following the GTFS proposal):
+
+```
+[
+  :trip_id,
+  :stop_sequence,
+  :occupancy_status,
+  :occupancy_percentage,
+  :valid_days,
+  :start_date,
+  :end_date,
+  :start_time
+]
+```
+
+- The `:trip_id` can be looked up using a trimmed version of `cTrainNo` which is equivalent to a
+  trip's GTFS `trip_short_name`.
+- `:stop_sequence` will be `nil`
+- `:occupancy_status` can in theory be one of `:empty`, `:many_seats_available`,
+  `:few_seats_available`, `:standing_room_only`, `:crushed_standing_room_only`, `:full`, or
+  `:not_accepting_passengers`, but for the Keolis data will only use 3 of them, with the mapping
+  based on `MedianDensityFlag` of `0` to `:many_seats_available`, `1` to `:few_seats_available`,
+  and `2` to `:full` (unless dotcom already has a different mapping of GTFS terms to bowling
+  pins?).
```

Comment:
Sure, I'll say that. But since Cristen confirmed the values and the prompt for this section is "it is reasonably clear how the feature would be implemented", I figure might as well mention the actual values here, too.